pkg/querier/blocks_store_queryable.go

```diff
@@ -1116,6 +1116,13 @@ func isRetryableError(err error) bool {
 	switch status.Code(err) {
 	case codes.Unavailable:
 		return true
+	// Client side connection closing, this error happens during store gateway deployment.
+	// https://github.com/grpc/grpc-go/blob/03172006f5d168fc646d87928d85cb9c4a480291/clientconn.go#L67
+	case codes.Canceled:
+		return strings.Contains(err.Error(), "grpc: the client connection is closing")
+	// TODO(yeya24): change Thanos to use ResourceExhausted for chunk pool error.
+	case codes.Unknown:
+		return strings.Contains(err.Error(), "allocate chunk bytes")
```

Comment:
This will retry all SG requests that failed due to chunk pool exhausted? Is there a way for us to identify request that actually fetched too much chunks, and exclude that request from this retry? Otherwise this could just increase the impact from a heavy query.

Reply:
Good point. I removed retry on chunk pool for now after discussing with @alanprot offline. see if there is a better way to do that.

My 2cents, if it is a single request that exhaust the pool, there should be a correct limit to throttle it to not retry. That's the correct way to handle it.
Otherwise, all requests should be valid to retry. If we retry and caused more pool exhaustion, it is the problem of requests scheduling or we need scale up to reduce load.

Reply:
Nice, yes I agree